ShopCart/Program1.js

```diff
@@ -5,67 +5,95 @@ app.controller('TestCtrl', [ '$scope',
     }
 ]);
 
-app.directive('testDirec', function () {
-    
-    
-    function linkingFunction(scope, element, attrs) {
-    }
+app.directive('product', function () {
+ 
+    function linkingFunction(scope, element, attrs) {}
 
     function controller($scope) {
-        
-        $scope.productList=[
+              $scope.productList=[
         {
             id:0,
             name:"product0",
             price:34,
-            qty:4,
+            qty:1,
         },
         {
             id:1,
             name:"product1",
             price:93,
-            qty:4,
+            qty:1,
         },
         {
             id:3,
             name:"product3",
             price:4,
-            qty:23,
+            qty:1,
         }
-        ];
-        ///list of items on the cart        
-        $scope.cartList=[
-            {
-            id:1,
-            name:"product1",
-            price:93,
-            qty:2,
-            }];
-                
-        //adds to shopping cart
-        $scope.addToCart=function(product){
-            var idSearch="qty"+product.id;
-            var qty=Number(document.getElementById(idSearch).value);
+        ];   
+      $scope.getQty=function(product){
+           var idSearch="qty"+product.id;
+           $scope.qty= Number(document.getElementById(idSearch).value);   
```

Comment:
Normally you would use parseInt or parseFloat as it shows you have an understanding and expectation.  Using Number seems to loose.